### PR TITLE
Copy project holds when cloning

### PR DESCRIPTION
**Pull request by cpeel**
When cloning a project, clone the project holds as well. This is [Task 1993](https://www.pgdp.net/c/tasks.php?task_id=1993). This was super easy now that we hold the project data in a Project object so metadata like this doesn't get in the way.

Testable in the [clone-project-holds](https://www.pgdp.org/~cpeel/c.branch/clone-project-holds/) sandbox.

### CODE REVIEW

File: tools/project_manager/editproject.php [hunk 1]
```diff
@@ -304,6 +307,7 @@ public function set_from_post()
         }
 
         $this->original_marc_array_encd = @$_POST['rec'];
+        $this->hold_states = $this->unserialize(@$_POST['hold_states']) ?? [];
```
Comment (chrismiceli) [inline tools/project_manager/editproject.php:310]: any reason we shouldn't trust the client with whatever hold states it sends here? I can't think of one to be honest.

Comment (cpeel) [reply tools/project_manager/editproject.php]: Well, we should never trust user input 😁 

We correctly escape values before putting them into the database, so there is low risk here, but validation is always better and I need to do this when I add this to the API too. Thanks for calling it out.

**Review by srjfoo — APPROVED** [review]

File: tools/project_manager/editproject.php [hunk 2]
```diff
@@ -560,4 +573,18 @@ public function normalize_spaces()
         $this->project->clearance = preg_replace('/\s+/', ' ', trim($this->project->clearance));
         $this->project->extra_credits = preg_replace('/\s+/', ' ', trim($this->project->extra_credits));
     }
+
+    private function serialize($value)
+    {
```
Comment (70ray) [inline tools/project_manager/editproject.php:578]: Could not redefining a function with the same name lead to confusion for future generations?

**Review by 70ray — APPROVED** [review]

File: tools/project_manager/editproject.php (hunk 2 above)
Comment (cpeel) [reply tools/project_manager/editproject.php:578]: It's a class function so it isn't overriding the built-in (not that PHP supports that like python does). The only way to call this function is to use the object.

**Pull request merged**